lib/sqlalchemy/dialects/mysql/expression.py

```diff
@@ -0,0 +1,138 @@
+from functools import wraps
```

Comment:
But i'm -1 to complicate the implementation and violate SRE principe
I'm -1 in reinvention the wheel in query syntax. SA is very useful because it allows to write SQL-like python code. Why not to allow write this thing in SQL style?

Reply:
you are free to include a match() function in dialects.mysql:

```
def match(*cols, **flags...):
   return cols[0].match(... flags, mysql_additional_cols=cols[1:])
```

then you have the same interface.   the parts I disagree with are 1. new clauseelement class  and 2. using properties as generative methods.    you don't need 1 at all.    2. I think is much simpler with kwargs.    no SRE is being violated, the mysql visit_match is where this is entirely encapsulated and that's in fact why I prefer this.   the proposed `ClauseElementBatch` currently has no other purpose and is only an implementation detail.

Reply:
The `ClauseElementBatch` can be used in [MSSQL:contains](https://docs.microsoft.com/en-us/sql/t-sql/queries/contains-transact-sql?view=sql-server-ver15) without any change

Reply:
heya.   did we know that earlier?     that would affect the API we do here because then yes, maybe we do want "match()" to be in sqlalchemy.sql.        This PR would need to be revised to support both of those and we would need tests and such.   ``BooleanClauseList`` is fine for a container of left hand side expressions.

Reply:
```
>>> b = BooleanClauseList(U.id, U.current_snapshot_id)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "***/venv/lib/python3.7/site-packages/sqlalchemy/sql/elements.py", line 2206, in __init__
    "BooleanClauseList has a private constructor"
NotImplementedError: BooleanClauseList has a private constructor
```

How to use it?

Reply:
use the _construct_raw() method.   take a look at the source.

Reply:
```
_construct_raw(operator, clauses=None)
```

`sqlalchemy.sql.operators.or_` => `operator.or_`

What operator should be used? `,`?

Reply:
> that would affect the API we do here because then yes, maybe we do want "match()" to be in sqlalchemy.sql

PostgreSQL can not match multiple columns
It looks [Oracle](https://docs.oracle.com/cd/B19306_01/text.102/b14218/csql.htm#i997503) can not match multiple columns

I think it's dialect specific behavior

Reply:
operators.comma_op


here's the whole feature in a demo 

```python
from sqlalchemy import Column
from sqlalchemy import create_engine
from sqlalchemy import Index
from sqlalchemy import MetaData
from sqlalchemy import select
from sqlalchemy import String
from sqlalchemy import Table
from sqlalchemy.sql import operators
from sqlalchemy.sql.elements import BooleanClauseList


def match(cols, expr):
    b = BooleanClauseList._construct_raw(operators.comma_op, clauses=cols)
    # this can be added as a kwarg to _construct_raw
    b.group = False
    return b.match(expr)


e = create_engine("mysql://scott:tiger@localhost/test", echo=True)
m = MetaData()
t = Table("t", m, Column("a", String(50)), Column("b", String(50)))
Index(
    "t_a_ix",
    t.c.a,
    t.c.b,
    mysql_prefix="FULLTEXT",
    mariadb_prefix="FULLTEXT",
)
with e.connect() as conn:
    t.drop(conn, checkfirst=True)
    t.create(conn)
    conn.execute(
        t.insert(),
        [{"a": "beans", "b": "peas"}, {"a": "carrots", "b": "beans"}],
    )

    rows = conn.execute(select(t).where(match([t.c.a, t.c.b], "beans")))

    assert set(rows) == {("beans", "peas"), ("carrots", "beans")}
```

Reply:
done